### PR TITLE
Add target type to string function #246

### PR DESCRIPTION
**Pull request by abhakta-47**
Implements STUMPLESS_FOREACH_TARGET_TYPE macro and related tests as in #246

### CODE REVIEW

File: include/stumpless/target.h [hunk 1]
```diff
@@ -1535,6 +1576,30 @@ vstumpless_trace_message( struct stumpless_target *target,
...
+ * This function is safe to call from threads that may be asynchronously
+ * cancelled.
+ *
+ * @param target_type The target_type to get the string from.
+ *
+ * @return The string representation of the given target_type.
+ */
+STUMPLESS_PUBLIC_FUNCTION
+const char *
+stumpless_get_target_type_string( enum stumpless_target_type target );
```
Comment (goatshriek) [inline include/stumpless/target.h:1601]: As noted [here](https://github.com/goatshriek/stumpless/blob/latest/docs/development.md#adding-new-functions), add this function to both the `tools/check_headers/stumpless.yml` manifest as well as `src/windows/stumpless.def`.

File: include/stumpless/target.h [hunk 2]
```diff
@@ -45,16 +45,57 @@
...
+/**< add to the Windows Event Log */
+# define STUMPLESS_WINDOWS_EVENT_LOG_TARGET_VALUE 7
+
+/**
+ * A macro function that runs the provided action once for each target_type,
+ * providing the symbol and value. The action must take two arguments, the
+ * first being the symbol name of the target_type, and the second the numeric
+ * value of the target_type.
+ */
+#  define STUMPLESS_FOREACH_TARGET_TYPE( ACTION )\
```
Comment (goatshriek) [inline include/stumpless/target.h:79]: Add the name of this macro function to the header check manifest file as well.

**Review by goatshriek — CHANGES_REQUESTED** [review]
Looks like this is mostly there! You just need to add the missing generator header noted by the header check tool, as well as make the adjustments listed in this review (which should resolve the other CI failures).

File: include/stumpless/target.h (hunk 1 above)
Comment (abhakta-47) [reply include/stumpless/target.h:1601]: for stumpless.def I am putting @177.

File: include/stumpless/target.h (hunk 2 above)
Comment (abhakta-47) [reply include/stumpless/target.h:79]: by header check manifest you mean `tools/check_headers/stumpless.yml` or `tools/check_headers/stumpless_private.yml`, `src/windows/stumpless.def` these files ?

actually STUMPLESS_FOREACH_SEVERITY is not added to any such files.

Comment (goatshriek) [reply include/stumpless/target.h:79]: Only to the `tools/check_headers/stumpless.yml` file. The severity version isn't their either as it was missed in that change, feel free to add it as well.

File: include/stumpless/target.h (hunk 1 above)
Comment (goatshriek) [reply include/stumpless/target.h:1601]: You'll need to do a fresh fetch from git to refresh your branch from the latest one, or you will encounter merge conflicts. The next available ordinal is currently 193.

**Comment by goatshriek** [comment]
Everything should be good to go, just add `stumpless/generator.h` to `include/stumpless/target.h` to resolve the header check error.

**Pull request merged**